app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/RefreshController.kt

```diff
@@ -387,12 +393,15 @@ class RefreshController @Inject constructor(
 
         apiResponse.data?.messages?.let { messages ->
 
+            var inboxUnreadCount: Int? = null
             write {
                 val upToDateFolder = getUpToDateFolder(folder.id)
                 val isConversationMode = localSettings.threadMode == ThreadMode.CONVERSATION
                 val allImpactedThreads = createThreads(scope, upToDateFolder, messages, isConversationMode).also { threads ->
-                    val foldersIds = (if (isConversationMode) threads.map { it.folderId }.toSet() else emptySet()) + folder.id
-                    foldersIds.forEach { refreshUnreadCount(id = it, realm = this) }
+                    inboxUnreadCount = updateFoldersUnreadCount(
+                        foldersIds = (if (isConversationMode) threads.map { it.folderId }.toSet() else emptySet()) + folder.id,
```

Comment:
```suggestion
                        foldersIds = (if (isConversationMode) threads.mapTo(mutableSetOf()) { it.folderId } else emptySet()) + folder.id,
```

Reply:
Fixed in #2076

Reply:
Ideally I'd prefer to not introduce suboptimal code then correct it later on, but everything's already approved so to gain time, I'll approve this one like this and everything can be merged directly